### PR TITLE
S12 product page suggested products

### PR DESCRIPTION
**Pull request by APaltrow**
- Added Product suggestions UI;
- Added Carousel UI and logic;
- Added UI responsiveness;
- Added logic for suggestions;
- Added Skeleton loaders for suggestions;

### CODE REVIEW

File: src/hooks/useCarousel.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const onSlidePrevClick = () => {
+    setOffset((prev) => {
+      const newOffset = prev - carouselStep;
+      const isContentEnd = newOffset + getContainerWidth() >= getContentWidth();
+
+      if (newOffset > CarouselConstants.INITIAL_OFFSET) {
+        setPrevVisible(true);
+      } else {
+        setPrevVisible(false);
+      }
```
Comment (Yuliia143) [inline src/hooks/useCarousel.ts]: setPrevVisible(newOffset > CarouselConstants.INITIAL_OFFSET) 
or
some variable = newOffset > CarouselConstants.INITIAL_OFFSET
setPrevVisible(that variable)

File: src/hooks/useCarousel.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+      if (isContentEnd) {
+        setNextVisible(false);
+      } else {
+        setNextVisible(true);
+      }
```
Comment (Yuliia143) [inline src/hooks/useCarousel.ts]: Duplicates

File: src/hooks/useCarousel.ts (hunk 1 above)
Comment (APaltrow) [reply src/hooks/useCarousel.ts]: Removed unnecessary conditions.

File: src/hooks/useCarousel.ts (hunk 2 above)
Comment (APaltrow) [reply src/hooks/useCarousel.ts]: Removed unnecessary conditions.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**